### PR TITLE
fix(kuma-dp): fix incorrect dataplane name due to mangled env vars

### PR DESCRIPTION
**Pull request by bartsmykla**
Kubernetes workloads may have their environment variables mangled by other components outside of our control. This can cause the `POD_NAME` or `POD_NAMESPACE` variables to be defined after the `KUMA_DATAPLANE_NAME` variable, which results in an incorrect dataplane name.

This commit fixes this issue by checking for the `POD_NAME` and `POD_NAMESPACE` variables directly during Kuma dataplane runtime initialization if `KUMA_DATAPLANE_NAME` is empty or contains the characters `$(){}`, which suggests that the dependent variables were not present before its definition. This involved adding a `PostProcess` method to the `Config` interface.

### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues
  - there is no relevant issue
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS
  - it won't
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s)
  - unit tests were added
  - manually tested on local k3d cluster
  - Don't forget `ci/` labels to run additional/fewer tests
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)?
  - there is no need
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label)
  - it does

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/config/app/kuma-dp/config.go [hunk 1]
```diff
@@ -116,8 +131,40 @@ type Dataplane struct {
...
+		if !ok {
+			return nil
+		}
+
+		podNamespace, ok := os.LookupEnv("POD_NAMESPACE")
+		if !ok {
+			return nil
+		}
+
+		d.Name = fmt.Sprintf("%s.%s", podName, podNamespace)
```
Comment (lahabana) [inline pkg/config/app/kuma-dp/config.go:146]: Should you redo validateMeshOrName here?

Comment (bartsmykla) [reply pkg/config/app/kuma-dp/config.go:146]: addressed

**Review by lahabana — COMMENTED** [review]
missing zoneProxy helm chart configuration change.

**Review by lahabana — APPROVED** [review]
Approved! THough I'm not sure we should backport this as it changes quite a lot.

**Pull request merged**